sdk/storage/azure-storage-blob-changefeed/azure/storage/blob/changefeed/_models.py

```diff
@@ -214,10 +218,19 @@ def __next__(self):
                 event = next(shard)
                 segment_events.append(event)
                 self.shards.append(shard)
-                self.cursor['shard_cursors'][shard.shard_path] = shard.cursor
-                self.cursor['cur_shard'] = shard.shard_path
             except StopIteration:
-                self.cursor['shard_cursors'][shard.shard_path] = "EOF"
+                pass
+
+            # update cursor
+            is_shard_cursor_in_list = False
```

Comment:
using a list of shard cursor is less efficient than what we previously had.
previously it was a dict of shard path and shard cursor, so everytime we want to update a shard cursor we can get(shard_path).

Reply:
you can convert list to map once somewhere where cursor is supplied to public api and use map in internal implementation to speed up shard cursor lookup. Then when someone requests cursor you can convert map back into list before returning it.

Reply:
yeah I think I have some very similar code in Java

Reply:
added the dict internally